plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -778,6 +790,35 @@ public Object[][] unsupportedTimestampDataProvider()
         };
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testClickHouseDateTimeWithTimeZone(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        dateTimeWithTimeZoneTest(clickhouseDateTimeInputTypeFactory("datetime"), clickhouseDateTimeInputLiteralFactory(), trinoTimestampExpectedLiteralFactory())
+                .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.datetime_tz"));
+    }
+
+    private SqlDataTypeTest dateTimeWithTimeZoneTest(Function<ZoneId, String> inputTypeFactory, BiFunction<LocalDateTime, ZoneId, String> inputLiteralFactory, BiFunction<LocalDateTime, ZoneId, String> expectedLiteralFactory)
+    {
+        ZoneId utc = ZoneId.of("UTC");
+        SqlDataTypeTest tests = SqlDataTypeTest.create()
+                .addRoundTrip(format("Nullable(%s)", inputTypeFactory.apply(utc)), "NULL", TIMESTAMP_TZ_SECONDS, "CAST(NULL AS TIMESTAMP(0) WITH TIME ZONE)");
+
+        for (ZoneId zoneId : ImmutableList.of(utc, kathmandu)) {
+            tests.addRoundTrip(inputTypeFactory.apply(zoneId), "0", TIMESTAMP_TZ_SECONDS, expectedLiteralFactory.apply(epoch.atZone(utc).withZoneSameInstant(zoneId).toLocalDateTime(), zoneId))
+                    .addRoundTrip(inputTypeFactory.apply(zoneId), inputLiteralFactory.apply(afterEpoch, zoneId), TIMESTAMP_TZ_SECONDS, expectedLiteralFactory.apply(afterEpoch, zoneId))
```

Comment:
no test with time before epoch?

Reply:
[ClickHouse's DateTime documentation](https://clickhouse.com/docs/en/sql-reference/data-types/datetime) says this:

> Supported range of values: [1970-01-01 00:00:00, 2106-02-07 06:28:15].

"before epoch" is not supported.

Reply:
Add a comment mentioning this.

Reply:
Added.